rholang/examples/linking/packages/BalanceMap.rho

```diff
@@ -0,0 +1,75 @@
+export BalanceMap in {
+  import IMapApi in {
+    new 
+      IMapEmptyMap, IMapUpdated, IMapGetOrElse
+    in {
+      IMapApi("EmptyMap", IMapEmptyMap) | IMapApi("GetOrElse", IMapGetOrElse) |
+      IMapApi("Updated", IMapUpdated) | 
+      for(
+          emptyMap <- IMapEmptyMap; getOrElse <- IMapGetOrElse;
+          updated <- IMapUpdated
+      ){
+        
+        contract BalanceMap(return) = {
+          new query, sub, add, transfer, dataStore, api in {
+            emptyMap(dataStore) |
+            contract query(key, return) = {
+              for(map <- dataStore) {
+                dataStore(map) | getOrElse(key, map, 0, return)
+              }
+            } |
+            contract sub(key, amount, success) = {
+              match amount > 0 with true => {
+                new balanceCh in {
+                  for(map <- dataStore) {
+                    getOrElse(key, map, 0, balanceCh) | for(balance <- balanceCh) {
```

Comment:
I think we need to start specing out a formatter. For starters I think multi-line Procs that happen after a | should go to the next line. In other words,

```
getOrElse(...) |
for (...) {
   ...
}
```

Reply:
Maybe we should make a Rholang style guide just like we have for Scala?

I put the `for` on the same line because it consumes what it produced immediately prior to it. But I'm happy to adopt any particular style if one is decided upon by the team.

Reply:
I see - personally I find it easier to see when I know what is running in parallel.
```

A |
B |
C |
D 
```